### PR TITLE
MXFileStore: Make sure data is flushed to files on [MXFileStore close]

### PR DESCRIPTION
**Pull request by manuroe**
https://github.com/vector-im/riot-ios/issues/1797

Detected by the test checkRoomAccountDataTags

### CODE REVIEW

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m [hunk 1]
```diff
@@ -606,8 +597,35 @@ - (void)commit
...
 - (void)close
 {
+    NSLog(@"[MXFileStore] close: %tu pendingCommits", pendingCommits);
+
+    // Flush pending commits
+    if (pendingCommits)
+    {
+        [self saveDataToFiles];
+
+        dispatch_sync(dispatchQueue, ^(void){
```
Comment (giomfo) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: it should be `dispatch_async`, no?

Comment (manuroe) [reply MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: No, we want to synchronously close the store.

Comment (manuroe) [reply MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: No, we want to synchronously close the store.

Comment (manuroe) [reply MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: Well, sync or async is not really an issue. The method ends with a sync dispatch to ensure every dispatched task is done.
Using sync here will just ensure that all tasks are complete a bit before.

Comment (giomfo) [reply MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: ok

**Pull request merged**